streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingListenerForwardingBus.scala

```diff
@@ -17,17 +17,38 @@
 
 package org.apache.spark.streaming.scheduler
 
-import java.util.concurrent.atomic.AtomicBoolean
+import org.apache.spark.scheduler.{LiveListenerBus, SparkListener, SparkListenerEvent}
+import org.apache.spark.util.ListenerBus
 
-import org.apache.spark.Logging
-import org.apache.spark.util.AsynchronousListenerBus
+/**
+ * Wrap StreamingListenerEvent as SparkListenerEvent so that it can be posted to Spark listener bus.
+ */
+private[streaming] case class WrappedStreamingListenerEvent(
+    streamingListenerEvent: StreamingListenerEvent) extends SparkListenerEvent {
+
+  // TODO once SPARK-12140 is resolved this will be true as well
+  protected[spark] override def logEvent: Boolean = false
+}
 
-/** Asynchronously passes StreamingListenerEvents to registered StreamingListeners. */
-private[spark] class StreamingListenerBus
-  extends AsynchronousListenerBus[StreamingListener, StreamingListenerEvent]("StreamingListenerBus")
-  with Logging {
+/**
+ * A Streaming listener bus to forward events in WrappedStreamingListenerEvent to StreamingListeners
+ */
+private[streaming] class StreamingListenerForwardingBus(sparkListenerBus: LiveListenerBus)
+  extends SparkListener with ListenerBus[StreamingListener, StreamingListenerEvent] {
```

Comment:
I think it's fine that it extends `ListenerBus` since it does in fact share the same method signatures. I don't see how the threading assumptions are violated if we call `postToAll`. There's only 1 thread (the core listener bus), and that thread calls this synchronously.

Also it's not just the 3 lines. We also want to inherit the `postToAll` method because we want to post to all streaming listeners, right? The implementation also has some try catch logic in there so I think it's better this way.